src/app/features/data-policy/data-policy.page.ts

```diff
@@ -0,0 +1,45 @@
+import { Component } from '@angular/core';
+import { NavController } from '@ionic/angular';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { BehaviorSubject, fromEvent } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { BUBBLE_IFRAME_URL } from '../../shared/dia-backend/secret';
+import { NetworkService } from '../../shared/network/network.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'app-data-policy',
+  templateUrl: './data-policy.page.html',
+  styleUrls: ['./data-policy.page.scss'],
+})
+export class DataPolicyPage {
+  readonly networkConnected$ = this.networkService.connected$;
+  readonly dataPolicyUrl = `${BUBBLE_IFRAME_URL}/data_policy`;
+  readonly iframeLoaded$ = new BehaviorSubject(false);
+
+  constructor(
+    private readonly networkService: NetworkService,
+    private readonly navController: NavController
+  ) {}
+
+  ionViewDidEnter() {
+    this.processIframeEvents();
+  }
+
+  processIframeEvents() {
+    fromEvent(window, 'message')
+      .pipe(
+        tap(event => {
+          const postMessageEvent = event as MessageEvent;
+          if (postMessageEvent.data === 'iframe-on-load') {
+            this.iframeLoaded$.next(true);
+          }
+          if (postMessageEvent.data === 'iframeBackButtonClicked') {
+            this.navController.back();
+          }
+        }),
+        untilDestroyed(this)
+      )
+      .subscribe();
+  }
+}
```

Comment:
@tammyyang regarding improvement suggestions
- If we use `@HostListener` then it would look like `@HostListener('window:message', ['$event']) onMessage(event: Event) {
    console.log('Process the message!');
  }` to me using `fromEvent` (line: 30) is more clear and has some auto-completion. So I would ignore this suggestion.
- it's already marked as `readonly`, I would ignore this suggestion.
- a good point need to do this for all iframes inside the capture app 
  - [[Code Quality + Security] capture-app-ionic. Validate the origin of the message event (iframe postMessage), this is to ensure that the event is sent from a trusted source.](https://app.asana.com/0/1201016280880500/1203794276110740/f)
  - [[Code Quality + Security] capture-app-iframe. Validate the origin of the message event (iframe postMessage), this is to ensure that the event is sent from a trusted source.](https://app.asana.com/0/0/1203794276110743/f)
- Does not make sense.